include/nbl/video/IPhysicalDevice.h

```diff
@@ -70,191 +72,30 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
-            uint32_t           maxRayRecursionDepth;
-            uint32_t           maxShaderGroupStride;
-            uint32_t           shaderGroupBaseAlignment;
-            uint32_t           shaderGroupHandleCaptureReplaySize;
-            uint32_t           maxRayDispatchInvocationCount;
-            uint32_t           shaderGroupHandleAlignment;
-            uint32_t           maxRayHitAttributeSize;
-            
-            /* Nabla */
-            uint32_t maxBufferSize;
-            uint32_t maxOptimallyResidentWorkgroupInvocations = 0u; //  its 1D because multidimensional workgroups are an illusion
-            uint32_t maxResidentInvocations = 0u; //  These are maximum number of invocations you could expect to execute simultaneously on this device.
-            asset::IGLSLCompiler::E_SPIRV_VERSION spirvVersion;
-
-            // utility functions
-            // In the cases where the workgroups synchronise with each other such as work DAGs (i.e. `CScanner`),
-            // `workgroupSpinningProtection` is meant to protect against launching a dispatch so wide that
-            // a workgroup of the next cut of the DAG spins for an extended time to wait on a workgroup from a previous one.
-            inline uint32_t computeOptimalPersistentWorkgroupDispatchSize(const uint64_t elementCount, const uint32_t workgroupSize, const uint32_t workgroupSpinningProtection=1u) const
-            {
-                assert(elementCount!=0ull && "Input element count can't be 0!");
-                const uint64_t infinitelyWideDeviceWGCount = (elementCount-1ull)/(static_cast<uint64_t>(workgroupSize)*static_cast<uint64_t>(workgroupSpinningProtection))+1ull;
-                const uint32_t maxResidentWorkgroups = maxResidentInvocations/workgroupSize;
-                return static_cast<uint32_t>(core::min<uint64_t>(infinitelyWideDeviceWGCount,maxResidentWorkgroups));
-            }
-        };
+        using SLimits = SPhysicalDeviceLimits;
+        using SFeatures = SPhysicalDeviceFeatures;
 
         struct SProperties
         {
-            /* Vulkan Core 1.0 */
+            /* Vulkan 1.0 Core  */
             APIVersion  apiVersion;
-            // uint32_t driverVersion;
-            // uint32_t vendorID;
-            // uint32_t deviceID;
+            uint32_t    driverVersion;
+            uint32_t    vendorID;
+            uint32_t    deviceID;
             E_TYPE      deviceType;
-            // char     deviceName[VK_MAX_PHYSICAL_DEVICE_NAME_SIZE];
-            // uint8_t  pipelineCacheUUID[VK_UUID_SIZE];
-            SLimits     limits; // Contains Limits on Vulkan Core 1.0, 1.1, 1.2 and extensions
+            char        deviceName[VK_MAX_PHYSICAL_DEVICE_NAME_SIZE];
+            uint8_t     pipelineCacheUUID[VK_UUID_SIZE];
+            SLimits     limits; // Contains Limits on Vulkan 1.0 Core , 1.1, 1.2 and extensions
             // VkPhysicalDeviceSparseProperties    sparseProperties;
```

Comment:
delete the comment, we have those (commented out) in `SLimits` dont we?

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/357/files#r887850184

Reply:
Yea it's a bit weird, the main reason it's there is because I wanted it to look like `VkPhysicalDeviceProperties` but I can move it now.

https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceProperties.html